cpp/src/gandiva/precompiled/timestamp_arithmetic.cc

```diff
@@ -172,6 +172,57 @@ TIMESTAMP_DIFF(timestamp)
     return millis + TO_MILLIS * static_cast<gdv_##TYPE>(count);          \
   }
 
+#define ADD_DAY_TIME_INTERVAL_TO_TIMESTAMP(TYPE, NAME, TO_MILLIS)              \
+  FORCE_INLINE                                                                 \
+  gdv_timestamp NAME##_day_time_interval_##TYPE(gdv_##TYPE millis,             \
+                                                gdv_day_time_interval count) { \
+    gdv_int64 day_interval_days = extractDay_daytimeinterval(count);           \
+    gdv_int64 day_interval_millis = extractMillis_daytimeinterval(count);      \
+    return static_cast<gdv_timestamp>(millis) +                                \
+           (day_interval_days * TO_MILLIS + day_interval_millis);              \
+  }
+
+#define SUB_DAY_TIME_INTERVAL_TO_TIMESTAMP(TYPE, NAME, TO_MILLIS)              \
+  FORCE_INLINE                                                                 \
+  gdv_timestamp NAME##_day_time_interval_##TYPE(gdv_##TYPE millis,             \
+                                                gdv_day_time_interval count) { \
+    gdv_int64 day_interval_days = extractDay_daytimeinterval(count);           \
+    gdv_int64 day_interval_millis = extractMillis_daytimeinterval(count);      \
+    return static_cast<gdv_timestamp>(millis) -                                \
+           (day_interval_days * TO_MILLIS + day_interval_millis);              \
+  }
```

Comment:
These macros are always returning a `gdv_timestamp`, but when the functions are receiving a `gdv_date64` they should return a variable of `gdv_date64` too, right?

Reply:
you are right, I'm sorry